scalyr_agent/agent_main.py

```diff
@@ -790,6 +794,12 @@ def __run(self, controller):
                             # Update the debug_level based on the new config.. we always update it.
                             self.__update_debug_log_level(new_config.debug_level)
 
+                        # see if we need to perform a garbage collection
+                        if gc_interval > 0 and current_time > next_gc_time:
```

Comment:
Why not just write the second clause as `current_time > last_gc_time + gc_interval`.  If you do that, then you do not need to do the little bit of odd dance you do below to make sure you properly update the `next_gc_time` based on the new `gc_interval`.

Reply:
Good idea!  Sometimes you miss the forest because of all the trees.  Update coming shortly.